### PR TITLE
Datasource: Fix - apply default query also to queries in new panels

### LINKED ISSUE #59689: Datasource: Default query not applied when creating new panels
**What happened**:
Default query defined by the data source using `getDefaultQuery` method is not applied to new queries when creating new panels.

**What you expected to happen**:

They should be applied (the way they are when creating new queries in an existing panel).

### PR DESCRIPTION
**Pull request by sunker**
**What is this feature?**

This PR attempts to fix a problem @mmandrus found in [this](https://github.com/grafana/grafana/pull/57703) PR, where the default query is only applied when new query rows are added but not when new panels are created. 

Will add tests if this turns out to be the right approach. 

**Why do we need this feature?**

This feature is useful because it allows datasources to define a default query without having to call `onChange` the first time the editor is rendered. 

**Who is this feature for?**

Plugin developers. 

Fixes #59689
Related to https://github.com/grafana/grafana/pull/49581
cc @idastambuk

### CODE REVIEW

File: public/app/core/utils/query.ts [hunk 1]
```diff
@@ -9,6 +9,27 @@ export const getNextRefIdChar = (queries: DataQuery[]): string => {
   }
 };
 
+export function queryIsEmpty(query: DataQuery): boolean {
```
Comment (sunker) [inline public/app/core/utils/query.ts:12]: This check does not feel future proof as it doesn't take new props into consideration. Any ideas of how to make this more robust?

File: public/app/core/utils/query.ts [hunk 2]
```diff
@@ -9,6 +9,27 @@ export const getNextRefIdChar = (queries: DataQuery[]): string => {
...
+  const schema: Record<keyof DataQuery, string> = {
+    refId: 'string',
+    hide: 'boolean',
+    key: 'string',
+    queryType: 'string',
+    datasource: 'object',
+  };
+
+  for (const key in query) {
+    if (key === 'label') {
```
Comment (sunker) [inline public/app/core/utils/query.ts:22]: Also seems like the query runner adds this `label` prop to the query. I have no idea where this is coming from.

File: public/app/core/utils/query.ts (hunk 1 above)
Comment (torkelo) [reply public/app/core/utils/query.ts:12]: is this not the same as just doing Object.keys(query).length === 0 ?

Comment (sunker) [reply public/app/core/utils/query.ts:12]: Currently this function is checking if the query has prop that is not defined in the `DataQuery` interface. So basically if it the data source has called `onChange` for the given query.

**Review by torkelo — APPROVED** [review]

**Pull request merged**